healthz/healthz.proto

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Generic Network Operation Interface, GNOI, defines a set of RPC's used for
+// the operational aspects of network targets. These services are meant to be
+// used in conjunction with GNMI for all target state and operational aspects
+// of a network target. The gnoi.system.Service is the only mandatory vendor
+// implementation.
+
+syntax = "proto3";
+
+
+package gnoi.healthz;
+
+import "github.com/openconfig/gnoi/types/types.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/openconfig/gnoi/healthz";
+
+option (types.gnoi_version) = "1.0.0";
+
+// The Healthz service provides access to the status of a path on the
+// system. Addtitionally it allows the implementor to provide path specific
+// diagnositic data into the status return.
+//
+// Healthz is expected to work in conjunction with the component OC model.
+service Healthz {
+  // Get will get health status for a gNMI path.  If no status is available for
+  // the requested path an error will be returned.
+  rpc Get(GetRequest) returns (GetResponse) {}
```

Comment:
Golang gRPC implementation has a problem with multiple services using the same name of a RPC call. It would help if it were changed to, say, GetHelthz().

Reply:
could you point me at an example of this that seems fundamentally broken and i would to someone on the gRPC / go teams to comment

If i register two services on to the mux - they better be namespaced by service

Obviously trying to register to services with the same name should fail

Reply:
talked internally this is really just a design possible issue where registering multiple services to the server struct may lead to having two service rpc methods trying to registered against that single struct which cannot provide overloading in go - the pattern should be that services are implemented via concrete structs and the server should be a composite of those service structs